reactioncommerce/reaction
# [Fix]: Migrate existing tag nav to new navigation tree structure

Resolves #4881 
Impact: **minor**  
Type: **bugfix**

## Issue
With the merge of #4683, a new API for defining and loading navigation trees was introduced. As part of migration 50, a default navigation tree was created with a single "Home" item. The migration should have also converted the existing tag nav items to the new tree.

## Solution
This PR updates migration 50 to migrate the existing tag nav to the new tree structure.

## Breaking changes
None

## Testing
1. Login to Reaction and update the tag nav to look like the demo storefront: https://storefront.demo.reactioncommerce.com/
2. In the database, set Migrations.version back to 49 and restart the app
3. Run this query in Graphiql and confirm the tree structure matches what you see in the Reaction app (Compare the content -> value field)
```
{
  primaryShop {
    defaultNavigationTree(language: "en"){
      _id
      shopId
      name
      items {
        navigationItem {
          data {
            ...NavigationItemFields
          }
        }
        items {
          navigationItem {
            data {
              ...NavigationItemFields
            }
          }
          items {
            navigationItem {
              data {
                ...NavigationItemFields
              }
            }
          }
        }
      }
    }
  }
}

fragment NavigationItemFields on NavigationItemData {
  content {
    language
    value
  }
  classNames
  url
  isUrlRelative
  shouldOpenInNewWindow
}
```
